### PR TITLE
worktree: minor speedup for `doAddDirectory`

### PR DESCRIPTION
**Pull request by ThinkChaos**

### CODE REVIEW

**Comment by pjbgf** [comment]
Please add some tests to ensure we are not adding a regression here.

File: worktree_status.go [hunk 1]
```diff
@@ -299,23 +301,7 @@ func (w *Worktree) doAddDirectory(idx *index.Index, s Status, directory string,
...
-		return false
-	}
-
-	for i := 0; i < len(ds); i++ {
-		if ps[i] != ds[i] {
-			return false
-		}
-	}
-	return true
+	return directory == "." || strings.HasPrefix(path, directory+"/")
```
Comment (pjbgf) [inline worktree_status.go:304]: Replace `"/` with `os.PathSeparator`.

**Comment by ThinkChaos** [comment]
Added 2 tests:
- check adding a dir with trailing slash works
- check adding `.` works

Failures are unrelated (network errors).
Will force push to squash and retry the CI.

**Comment by pjbgf** [comment]
@ThinkChaos thank you for adding the tests. Please rebase and reword the commit based on our [contributing guidelines](https://github.com/go-git/go-git/blob/master/CONTRIBUTING.md#format-of-the-commit-message).

File: worktree_test.go [hunk 2]
```diff
@@ -1488,6 +1488,96 @@ func (s *WorktreeSuite) TestAddRemovedInDirectory(c *C) {
 	c.Assert(file.Staging, Equals, Unmodified)
 }
 
+func (s *WorktreeSuite) TestAddRemovedInDirectoryWithTrailingSlash(c *C) {
```
Comment (AriehSchneier) [inline worktree_test.go:1491]: nit: these 2 tests are almost identical, why not do them in a loop, eg:
```
	cases := []struct {
		name       string
		dir        string
		goStatus   string
		jsonStatus string
	}{
	...
	}
```

Comment (ThinkChaos) [reply worktree_test.go:1491]: I followed the precedent from the other tests, and didn't want to spend time refactoring them all. If it's fine with you, I'd rather leave it as is.

Comment (AriehSchneier) [reply worktree_test.go:1491]: Looks like I can't mark this comment as resolved, it was just a `nit` so all good.

**Comment by pjbgf** [comment]
Please reword the commit to `worktree: minor speedup for doAddDirectory`.

**Pull request merged**